sdk/servicebus/Azure.Messaging.ServiceBus/src/Transports/Amqp/ServiceBusSenderExtensions.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.ServiceBus.Transports.Amqp
+{
+    /// <summary>
+    /// These extensions allow additional functionality for the <see cref="ServiceBusClient"/>.
+    /// </summary>
+    public static class ServiceBusSenderExtensions
+    {
+        internal static ServiceBusMessage CreateAmqpDataMessage(IEnumerable<ReadOnlyMemory<byte>> data) =>
+            new ServiceBusMessage(new AmqpTransportBody { BodyType = AmqpBodyType.Data, Data = data });
+
+        /// <summary>
+        /// Creates a <see cref="ServiceBusMessage"/> with the AMQP Data body type.
+        /// </summary>
+        /// <param name="sender">The sender for the message. This is unused to create the message.</param>
+        /// <param name="data">The message body to provide to the AQMP Data body.</param>
+        /// <returns>A new <see cref="ServiceBusMessage"/>.</returns>
+        public static ServiceBusMessage CreateAmqpDataMessage(this ServiceBusSender sender, IEnumerable<ReadOnlyMemory<byte>> data) =>
+            CreateAmqpDataMessage(data);
+
+        internal static ServiceBusMessage CreateAmqpSequenceMessage(IEnumerable<IList> sequence) =>
```

Comment:
nit: can we just collapse these into the extension methods?

Reply:
I could... but the internal code would be strange. I'd be providing `null` for sender when I end up calling it. Remember, the public one is an extension method for a type in which the type itself isn't used.

Reply:
Ah, nevermind - forgot this is used internally.